### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by Chris-Wolfgang**
## Description

Fixes/Complete # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update
- [ ] Refactor

## How Has This Been Tested?

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/Wolfgang.Extensions.IEnumerable/IEnumerableExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    public static bool None<T>(this IEnumerable<T> source, Func<T, bool> predicate)
+        => !source.Any(predicate);
+
+
+
+    /// <summary>
+    /// Creates a new IEnumerable{T} containing the elements from source in a random order
+    /// </summary>
+    /// <typeparam name="T">The type of the elements of source.</typeparam>
+    /// <param name="source">An IEnumerable{T} whose will be randomly ordered.</param>
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply src/Wolfgang.Extensions.IEnumerable/IEnumerableExtensions.cs]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/Chris-Wolfgang/IEnumerable-Extensions/pull/1#discussion_r2702926238)

File: src/Wolfgang.Extensions.IEnumerable/IEnumerableExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    /// <returns>A new IEnumerable{T} containing the elements from source in a random order.</returns>
+    /// <exception cref="ArgumentNullException">source is null.</exception>
+    public static IEnumerable<T> Shuffle<T>(this IEnumerable<T> source)
+        => source
+            .OrderBy(_ => Guid.NewGuid())
+            .ToList();
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply src/Wolfgang.Extensions.IEnumerable/IEnumerableExtensions.cs]: @copilot Create a benchmark to compare the current implementation and this suggestion

File: src/Wolfgang.Extensions.IEnumerable/IEnumerableExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    ///     {
+    ///         var numbers = new[] { 1, 2, 3, 4 };
+    ///
+    ///         Console.WriteLine (number.None(n => n % 3)); // Prints false
+    ///         Console.WriteLine (number.None(n => n % 5)); // Prints true
+    ///     }
+    /// 
+    /// </example>
+    public static bool None<T>(this IEnumerable<T> source, Func<T, bool> predicate)
+        => !source.Any(predicate);
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply src/Wolfgang.Extensions.IEnumerable/IEnumerableExtensions.cs]: @copilot change from expression body to standard body and check for null. Also create a test to verify that passing null for the predicate throws ArgumentNullException
